bundles/org.openhab.binding.ism8/src/main/java/org/openhab/binding/ism8/internal/Ism8Handler.java

```diff
@@ -0,0 +1,158 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ism8.internal;
+
+import java.io.IOException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.ism8.server.DataPointChangedEvent;
+import org.openhab.binding.ism8.server.IDataPoint;
+import org.openhab.binding.ism8.server.IDataPointChangeListener;
+import org.openhab.binding.ism8.server.Server;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Ism8Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Hans-Reiner Hoffmann - Initial contribution
+ */
+@NonNullByDefault
+public class Ism8Handler extends BaseThingHandler implements IDataPointChangeListener {
+    private final Logger logger = LoggerFactory.getLogger(Ism8Handler.class);
+
+    private @Nullable Ism8Configuration config;
+    private @Nullable Server server;
+
+    public Ism8Handler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        this.logger.debug("Ism8: Handle command = {} {}", channelUID.getId(), command);
+        Channel channel = getThing().getChannel(channelUID);
+        Server svr = this.server;
+        if (channel != null && svr != null) {
+            if (channel.getConfiguration().containsKey("id") && channel.getConfiguration().containsKey("write")
+                    && channel.getConfiguration().get("write").toString().equalsIgnoreCase("true")) {
+                int id = Integer.parseInt(channel.getConfiguration().get("id").toString());
+                this.logger.debug("Channel '{}' writting into ID '{}'", channel.getUID().getId(), id);
+                this.updateState(channelUID, new QuantityType<>(command.toString()));
+
+                IDataPoint dataPoint = null;
+                for (IDataPoint dp : svr.getDataPoints()) {
+                    if (dp.getId() == id) {
+                        dataPoint = dp;
+                        break;
+                    }
+                }
+
+                if (dataPoint != null) {
+                    try {
+                        svr.sendData(dataPoint.createWriteData(command));
+                    } catch (IOException e) {
+                        this.logger.warn("Writting to ISM DataPoint '{}' failed. '{}'", dataPoint.getId(),
+                                e.getMessage(), e);
```

Comment:
Logging the stack trace when the network fails seems a bit unreasonable.

Reply:
Changed to debug.